bundles/org.aposin.gem.core/src/org/aposin/gem/core/api/service/IServiceContainer.java

```diff
@@ -18,17 +18,40 @@
 import java.util.Collection;
 import java.util.Map;
 
+import org.aposin.gem.core.GemException;
 import org.aposin.gem.core.api.IRefreshable;
 import org.aposin.gem.core.api.config.GemConfigurationException;
 import org.aposin.gem.core.api.config.IConfigurable;
 import org.aposin.gem.core.api.service.launcher.IEnvironmentLauncherProvider;
 import org.aposin.gem.core.api.service.launcher.IFeatureBranchLauncherProvider;
+import org.aposin.gem.core.impl.service.DefaultGemSorter;
 
 /**
  * Container class for the services implemented by core and/or extensions.
  */
 public interface IServiceContainer extends IRefreshable, IConfigurable {
 
+    /**
+     * Gets the configured {@link IGemSorter}.
+     * </br>
+     * Default implementation checks for an optional unique gem-sorter.
+     * 
+     * @return sorter.
+     * 
+     * @throws GemException if more than one sorter is provided.
+     */
+    public default IGemSorter getGemSorter() {
+        final Collection<IGemSorter> sorters = getGemServices(IGemSorter.class);

```

Comment:
Why a Collection and not a List (or Set)?

Reply:
The API for `getGemServices` is already returning a collection; this change is not refactoring that API but just using it. I believe that it is because the implementation of the `IServiceContainer` is using `Map::values` to return the services. In any case, as the services are used mostly internally to provide hooks on GEM, it shouldn't be a problem and we can always change between collection implementations as required.